### PR TITLE
update to add ascent 0.5.0 release

### PR DESCRIPTION
**Pull request by cyrush**
update ascent, vtk-h, and vtkm to new releases. 
changes zfp variant default for conduit to false

Also includes fixes to metis download and flags passed to the mfem for conduit

### CODE REVIEW

File: var/spack/repos/builtin/packages/ascent/package.py [hunk 1]
```diff
@@ -28,19 +28,23 @@ def cmake_cache_entry(name, value, vtype=None):
...
     git      = "https://github.com/Alpine-DAV/ascent.git"
+    url      = "https://github.com/Alpine-DAV/ascent/releases/download/v0.5.0/ascent-v0.5.0-src-with-blt.tar.gz"
 
     maintainers = ['cyrush']
 
     version('develop',
             branch='develop',
-            submodules=True)
+            submodules=True,
+            preferred=True)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ascent/package.py]: We don't usually like to have non-checksummable versions be the default when possible

File: var/spack/repos/builtin/packages/ascent/package.py [hunk 2]
```diff
@@ -72,10 +77,15 @@ class Ascent(Package):
     # package dependencies
     ###########################################################################
 
-    depends_on("cmake@3.14:", type='build')
+    # use cmake 3.14, newest that provides proper cuda support
+    # and we have seen errors with cuda in 3.15
+    depends_on("cmake@3.14.1:3.14.99")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ascent/package.py]:
```suggestion
    depends_on("cmake@3.14.1:3.14.99", type='build')
```

File: var/spack/repos/builtin/packages/ascent/package.py [hunk 3]
```diff
@@ -85,6 +95,7 @@ class Ascent(Package):
     depends_on("python+shared", when="+python+shared")
     extends("python", when="+python+shared")
     depends_on("py-numpy", when="+python+shared", type=('build', 'run'))
+    depends_on("py-pip", when="+python+shared", type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ascent/package.py]: pip is rarely needed if you follow the standard `setup.py install` method

File: var/spack/repos/builtin/packages/ascent/package.py [hunk 4]
```diff
@@ -412,6 +431,13 @@ def create_host_config(self, spec, prefix, py_site_pkgs_dir=None):
             cfg.write("# vtk-h from spack\n")
             cfg.write(cmake_cache_entry("VTKH_DIR", spec['vtk-h'].prefix))
 
+            if "+cuda" in spec:
+                cfg.write(cmake_cache_entry("VTKm_ENABLE_CUDA", "ON"))
+                cfg.write(cmake_cache_entry("CMAKE_CUDA_HOST_COMPILER",
+                          ''.format(env["SPACK_CXX"])))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ascent/package.py]:
```suggestion
                          env["SPACK_CXX"]))
```

File: var/spack/repos/builtin/packages/metis/package.py [hunk 5]
```diff
@@ -17,13 +17,15 @@ class Metis(Package):
...
-    homepage = "http://glaros.dtc.umn.edu/gkhome/metis/metis/overview"
-    url      = "http://glaros.dtc.umn.edu/gkhome/fetch/sw/metis/metis-5.1.0.tar.gz"
-    list_url = "http://glaros.dtc.umn.edu/gkhome/fsroot/sw/metis/OLD"
+    #
+    # the previous metis website http://glaros.dtc.umn.edu/gkhome/metis/metis
+    # no longer exists. This is a github mirror that provides metis 5.1.0
+    #
+
+    homepage = "https://github.com/scivision/METIS/"
+    url      = "https://github.com/scivision/METIS/raw/master/metis-5.1.0.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/metis/package.py]: I wish there were a better permanent solution. Only having access to a single version is kind of sad.

File: var/spack/repos/builtin/packages/mfem/package.py [hunk 6]
```diff
@@ -439,8 +439,36 @@ def find_optional_library(name, prefix):
...
+            #  To resolve the issue, we simply join our own string with
+            #  the headers results (which are important b/c they handle
+            #  hdf5 paths when enabled).
+            ##################
+
+            # construct proper include path
+            conduit_include_path = join_path(conduit.prefix.include,
+                                             "conduit")
+            # add this path to the found flags
+            conduit_opt_flags = "-I{} {}".format(conduit_include_path,
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mfem/package.py]:
```suggestion
            conduit_opt_flags = "-I{0} {0}".format(conduit_include_path,
```
Needed for Python 2.6 support

File: var/spack/repos/builtin/packages/mfem/package.py [hunk 7]
```diff
@@ -439,8 +439,36 @@ def find_optional_library(name, prefix):
...
+            #  breaking build.
+            #
+            #  To resolve the issue, we simply join our own string with
+            #  the headers results (which are important b/c they handle
+            #  hdf5 paths when enabled).
+            ##################
+
+            # construct proper include path
+            conduit_include_path = join_path(conduit.prefix.include,
+                                             "conduit")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mfem/package.py]: You could just do `conduit.prefix.include.conduit`

File: var/spack/repos/builtin/packages/vtk-h/package.py [hunk 8]
```diff
@@ -28,49 +28,45 @@ def cmake_cache_entry(name, value, vtype=None):
...
-    variant("tbb", default=False, description="build tbb support")
+    variant("serial", default=True, description="build serial (non-mpi) libraries")
     variant("cuda", default=False, description="build cuda support")
     variant("openmp", default=(sys.platform != 'darwin'),
             description="build openmp support")
 
-    depends_on("cmake@3.8.2:", type='build')
+    # use cmake 3.14, newest that provides proper cuda support
+    # and we have seen errors with cuda in 3.15
+    depends_on("cmake@3.14.1:3.14.99")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/vtk-h/package.py]:
```suggestion
    depends_on("cmake@3.14.1:3.14.99", type='build')
```

File: var/spack/repos/builtin/packages/vtk-m/package.py [hunk 9]
```diff
@@ -22,11 +22,11 @@ class VtkM(CMakePackage, CudaPackage):
...
-    version('1.5.0', sha256="cd38957bb552e28b5197e4f738d5bb0b2c6f4025c3e17a66c8b19ee501273fbe")
-    version('1.4.0', sha256="8d83cca7cd5e204d10da151ce4f1846c1f7414c7c1e579173d15c5ea0631555a")
-    version('1.3.0', sha256="f88c1b0a1980f695240eeed9bcccfa420cc089e631dc2917c9728a2eb906df2e")
-    version('1.2.0', sha256="607272992e05f8398d196f0acdcb4af025a4a96cd4f66614c6341f31d4561763")
-    version('1.1.0', sha256="78618c81ca741b1fbba0853cb5d7af12c51973b514c268fc96dfb36b853cdb18")
+    version('1.5.0', sha256="b1b13715c7fcc8d17f5c7166ff5b3e9025f6865dc33eb9b06a63471c21349aa8")
+    version('1.4.0', sha256="60e1ce73a8c6beda8aed5f2d3ae670b6b0c78c068c6eff4ece769e6d719d5065")
+    version('1.3.0', sha256="72c2c8525a77a456fe0b6a1af0328dad6b9a688f402a3d3ebfa8942e0b5dba1a")
+    version('1.2.0', sha256="9103d954284661f6f03e5b18be6d7bc94254603e6abc8fce67f617f4ad325a0e")
+    version('1.1.0', sha256="a1746b1547d6fb901ea7d7ed50834e8832d6d41ddd497c84d02e1481100d43bc")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/vtk-m/package.py]: Why the checksum change?

File: var/spack/repos/builtin/packages/vtk-m/package.py [hunk 10]
```diff
@@ -91,7 +99,8 @@ def cmake_args(self):
 
             # logging support
             if "+logging" in spec:
-                if spec.satisfies('@:1.2.0'):
+                if spec.satisfies('@:1.2.0') and \
+                        spec['vtk-m'].version.string != 'master':
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/vtk-m/package.py]: Now that #1983 has been merged, this line should no longer be necessary. `master` will never satisfy `:1.2.0`

File: var/spack/repos/builtin/packages/vtk-m/package.py (hunk 9 above)
Comment (cyrush) [reply var/spack/repos/builtin/packages/vtk-m/package.py]: great catch, the 1.5.0 was wrong (had changed), I think I assumed my package.py  sha copies where correct based on the fact that 1.5.0 was wrong. That was not the case, I downloaded and checked them all, only 1.5.0 needs to be updated. I'll fix

File: var/spack/repos/builtin/packages/vtk-m/package.py (hunk 10 above)
Comment (cyrush) [reply var/spack/repos/builtin/packages/vtk-m/package.py]: thanks, I'll pull these out

File: var/spack/repos/builtin/packages/vtk-h/package.py (hunk 8 above)
Comment (cyrush) [reply var/spack/repos/builtin/packages/vtk-h/package.py]: thanks for catching, I'll fix

File: var/spack/repos/builtin/packages/metis/package.py (hunk 5 above)
Comment (cyrush) [reply var/spack/repos/builtin/packages/metis/package.py]: I agree, but I think this is the only option. The other site is completely gone. It has been gone for a bit

File: var/spack/repos/builtin/packages/ascent/package.py (hunk 4 above)
Comment (cyrush) [reply var/spack/repos/builtin/packages/ascent/package.py]: :-) thanks for catching, I'll fix

File: var/spack/repos/builtin/packages/ascent/package.py (hunk 3 above)
Comment (cyrush) [reply var/spack/repos/builtin/packages/ascent/package.py]: agree, it looks dubious -- but we need it for some things post spack install (testing and some use cases). It's very convenient so i'd like to keep it.

File: var/spack/repos/builtin/packages/ascent/package.py (hunk 2 above)
Comment (cyrush) [reply var/spack/repos/builtin/packages/ascent/package.py]: thanks for catching, I'll fix

**Comment by cyrush** [comment]
@adamjstewart I think i addressed everything, thanks for the feedback!

**Comment by cyrush** [comment]
is the codecov is relative to current develop? Are things are lower due to develop marching on? 

I plan to keep this branch as is, since I have spent many hours testing in CI, Docker Containers, on a llnl system, nersc's cori, and on my mac. I don't want to merge in spack develop to this branch, but if we need to do so I can create another branch (and then i guess PR) that merges develop. Just let me know.

**Comment by cyrush** [comment]
@adamjstewart - Sorry to bug -- I am sure you folks are working hard for the SC tutorial, any change we can get this merged? Would really like to tell folks at SC our latest and greatest is on spack develop.

File: var/spack/repos/builtin/packages/metis/package.py (hunk 5 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/metis/package.py]: Do we want to do the same thing to the `parmetis` package?

File: var/spack/repos/builtin/packages/vtk-m/package.py [hunk 11]
```diff
@@ -22,7 +22,7 @@ class VtkM(CMakePackage, CudaPackage):
     git      = "https://gitlab.kitware.com/vtk/vtk-m.git"
 
     version('master', branch='master')
-    version('1.5.0', sha256="cd38957bb552e28b5197e4f738d5bb0b2c6f4025c3e17a66c8b19ee501273fbe")
+    version('1.5.0', sha256="b1b13715c7fcc8d17f5c7166ff5b3e9025f6865dc33eb9b06a63471c21349aa8")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/vtk-m/package.py]: Did you investigate this checksum change?

Comment (cyrush) [reply var/spack/repos/builtin/packages/vtk-m/package.py]: I verified that is the proper checksum for that ver, obtained from the given url

File: var/spack/repos/builtin/packages/metis/package.py (hunk 5 above)
Comment (cyrush) [reply var/spack/repos/builtin/packages/metis/package.py]: Good call, we aren using parmetis and I am not an expert on testing it. 

I added a spack issue to help track the need for url update:
https://github.com/spack/spack/issues/13762

File: var/spack/repos/builtin/packages/vtk-m/package.py (hunk 11 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/vtk-m/package.py]: But why did the checksum change? Just want to make sure the source wasn’t hacked and the developers are aware of this checksum change

Comment (cyrush) [reply var/spack/repos/builtin/packages/vtk-m/package.py]: i am not sure, but i trust Kitware's official gitlab url.  Perhaps the retagged the release?

**Comment by cyrush** [comment]
@adamjstewart thanks for your feedback and help with this!

**Pull request merged**